hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/helpers/DatanodeIdYaml.java

```diff
@@ -0,0 +1,186 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.ozone.container.common.helpers;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.collections.MapUtils;
+import org.apache.hadoop.hdds.protocol.DatanodeDetails;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.Yaml;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * Class for creating datanode.id file in yaml format.
+ */
+public class DatanodeIdYaml {
+
+  private DatanodeIdYaml() {
+
+  }
+
+  /**
+   * Creates a yaml file using DatnodeDetails. This method expects the path
+   * validation to be performed by the caller.
+   *
+   * @param datanodeDetails {@link DatanodeDetails}
+   * @param path            Path to datnode.id file
+   */
+  public static void createDatanodeIdFile(DatanodeDetails datanodeDetails,
+                                          File path) throws IOException {
+    Preconditions.checkNotNull(path);
+    DumperOptions options = new DumperOptions();
+    options.setPrettyFlow(true);
+    options.setDefaultFlowStyle(DumperOptions.FlowStyle.FLOW);
+    Yaml yaml = new Yaml(options);
+
+    try (Writer writer = new OutputStreamWriter(
+        new FileOutputStream(path), "UTF-8")) {
+      yaml.dump(getDatanodeDetailsYaml(datanodeDetails), writer);
+    }
+  }
+
+  /**
+   * Read datanode.id from file.
+   */
+  public static DatanodeDetails readDatanodeIdFile(File path)
+      throws IOException {
+    Preconditions.checkNotNull(path);
+    DatanodeDetails datanodeDetails;
+    try (FileInputStream inputFileStream = new FileInputStream(path)) {
+      Yaml yaml = new Yaml();
+      DatanodeDetailsYaml datanodeDetailsYaml;
+      try {
+        datanodeDetailsYaml =
+            yaml.loadAs(inputFileStream, DatanodeDetailsYaml.class);
+      } catch (Exception e) {
+        throw new IOException("Unable to parse yaml file.", e);
+      }
+
+      DatanodeDetails.Builder builder = DatanodeDetails.newBuilder();
+      builder.setUuid(datanodeDetailsYaml.getUuid().toString())
+          .setIpAddress(datanodeDetailsYaml.getIpAddress())
```

Comment:
Oh I didn't even realize IP address was a part of this. We should probably not have it as IPs can change. Not related to your patch of course, we can fix it separately.

Reply:
Should I remove it from being serialized?

Reply:
But reading it back would set ip address to be empty. Best to handle as separate Jira. Do you mean it should a part of DatanodeDetails but be transient?

Reply:
Created HDDS-1480 to track.